.github/actions/test-dotnet/Merge-BlameHangDumps.ps1

```diff
@@ -0,0 +1,46 @@
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', 'Configuration', Justification = 'The Configuration parameter is in use.')]
+param ($Directory, $Configuration)
+
+$rootDirectory = (Resolve-Path $Directory)
+$blameHangDumpsName = 'BlameHangDumps'
+$dumpDirectory = (New-Item -Type Directory -Path $rootDirectory -Name $blameHangDumpsName)
+$testDirectoryPath = Join-Path $Directory 'test'
+$testDirectory = (Test-Path -Path $testDirectoryPath) ? (Resolve-Path $testDirectoryPath) : $rootDirectory
+
+# Save dotnet --info output.
+dotnet --info *> (Join-Path -Path $dumpDirectory.FullName -ChildPath 'dotnet.info')
+
+function ItemFilter($Item, $TestConfiguration)
+{
+    if ($Item.IsContainer)
+    {
+        return $False
+    }
+
+    $allow = (($PSItem.Name -like 'Sequence_*.xml') -or ($PSItem.Name -like '*_hangdump.dmp'))
+    if (!$allow -and $TestConfiguration)
```

Comment:
Please use `-not` instead of `!`. Only `-not` is recognized as a logical operator [in the documentation](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_operators?view=powershell-7.3#logical-operators) so `!` is basically an alias and we aren't supposed to use those. Same below.